link-rest/src/main/java/com/nhl/link/rest/runtime/processor/select/ParseRequestStage.java

```diff
@@ -57,26 +63,76 @@ public ProcessorOutcome execute(SelectContext<?> context) {
+                .limit(getLimit(request, protocolParameters));
 
+        context.setRawRequest(requestBuilder.build());
+    }
+
+    private CayenneExp getCayenneExp(LrRequest request, Map<String, List<String>> protocolParameters) {
+        return request != null && request.getCayenneExp() != null ?
+                request.getCayenneExp() :
+                expParser.fromString(ParameterExtractor.string(protocolParameters, PROTOCOL_KEY_CAYENNE_EXP));
+    }
+    private Sort getSort(LrRequest request, Map<String, List<String>> protocolParameters) {
+        return request != null && request.getSort() != null ?
+                request.getSort() :
+                sortParser.fromString(ParameterExtractor.string(protocolParameters, PROTOCOL_KEY_SORT));
+    }
+
+    private Dir getSortDirection(LrRequest request, Map<String, List<String>> protocolParameters) {
+        return request != null && request.getSortDirection() != null ?
+                request.getSortDirection() :
+                sortParser.dirFromString(ParameterExtractor.string(protocolParameters, PROTOCOL_KEY_DIR));
+    }
+
+    private MapBy getMapBy(LrRequest request, Map<String, List<String>> protocolParameters) {
+        return request != null && request.getMapBy() != null ?
+                request.getMapBy() :
+                mapByParser.fromString(ParameterExtractor.string(protocolParameters, PROTOCOL_KEY_MAP_BY));
+    }
+
+    private List<Include> getIncludes(LrRequest request, Map<String, List<String>> protocolParameters) {
+        return request != null && !request.getIncludes().isEmpty() ?
+                request.getIncludes() :
+                includeParser.fromStrings(ParameterExtractor.strings(protocolParameters, PROTOCOL_KEY_INCLUDE));
+    }
+
+    private List<Exclude> getExcludes(LrRequest request, Map<String, List<String>> protocolParameters) {
+        return request != null && !request.getExcludes().isEmpty() ?
+                request.getExcludes() :
+                excludeParser.fromStrings(ParameterExtractor.strings(protocolParameters, PROTOCOL_KEY_EXCLUDE));
+    }
+
+    private Start getStart(LrRequest request, Map<String, List<String>> protocolParameters) {
         int start = ParameterExtractor.integer(protocolParameters, PROTOCOL_KEY_START);
-        if (start >= 0) {
-            requestBuilder.start(new Start(start));
+        if (request != null && request.getStart() != null && request.getStart().getValue() >= 0) {
+            return request.getStart();
+        } else if (start >= 0) {
+            return new Start(start);
         }
 
+        return null;
+    }
+
+    private Limit getLimit(LrRequest request, Map<String, List<String>> protocolParameters) {
         int limit = ParameterExtractor.integer(protocolParameters, PROTOCOL_KEY_LIMIT);
-        if (limit >= 0) {
-            requestBuilder.limit(new Limit(limit));
+        if (request != null && request.getLimit() != null && request.getLimit().getValue() >= 0) {
```

Comment:
Should not be checking `request.getLimit().getValue() >= 0` . A null check is enough.